src/Vault.sol

```diff
@@ -344,6 +345,17 @@ contract Vault is SemiFungibleVault, ReentrancyGuard {
         assert(asset.transfer(_counterparty, idFinalTVL[id]));
     }
 
+    function sendFees(uint256 id) public onlyController marketExists(id) {
```

Comment:
this is wrong! this would mean that feeds will only be collected if `premium` is bigger than`collat`! 
Please stop reading from storage if you don't have to!

```solidity

    function sendFees(uint256 id) public onlyController marketExists(id) {
        uint256 assets = idFinalTVL[id];
        uint256 claimTVL = idClaimTVL[id];
        uint256 entitledShares = assets.mulDivUp(claimTVL, assets);
        if (entitledShares > assets) {
            uint256 premium = entitledShares - assets;
            uint256 feeValue = calculateWithdrawalFeeValue(premium, id);
            epochTreasuryFee[id] = feeValue;
            entitledShares = entitledShares - feeValue;
           if(feeValue > 0) { 
            assert(asset.transfer(treasury, feeValue));
           }
        }
    }
   ```

Reply:
this is a good improvement to my function

Reply:
However I dont understand the need to do 
```        uint256 entitledShares = assets.mulDivUp(claimTVL, assets);
```  
We are only and always were collecting fees if premium was bigger than collat

Reply:
collecting fees on all premiums no matter if bigger or smaller than collat!

Reply:
https://github.com/Y2K-Finance/Earthquake/blob/c4e1b11fa209fa31c704c5cc309698a13b6a9842/src/Vault.sol#L208

Reply:
we literally agreed when deploying the old contracts, fees are only taken on profits, check old code!  
are we changing that now?

Reply:
there is only premium if claimTVL is bigger than finalTVL , otherwise the Vault is at loss

Reply:
@MiguelBits didn't you write this code? If this would be the case we would only ever collect fees on Depeg or Hedge > Risk!  

https://github.com/Y2K-Finance/Earthquake/blob/c4e1b11fa209fa31c704c5cc309698a13b6a9842/src/Vault.sol#L208
`entitledShares != claimTVL` and `assets != finalTVL`

Please have a look at prev code implementation again!

Reply:
yes I did that's why I know what I am talking about and you dont...
``` entitledAmount = assets.mulDivUp(idClaimTVL[id], idFinalTVL[id]); ``` this line of code in https://github.com/Y2K-Finance/Earthquake/blob/c4e1b11fa209fa31c704c5cc309698a13b6a9842/src/Vault.sol#L369  
is there for the user to know HOW MUCH OF THE VAULT TVL in percentage BELONGS TO THE USER.  

Since you did not write the code you dont understand and I have to explain everything :
Controller TriggerEndEpoch set ClaimTVL is = risk tvl + hedge tvl :  https://github.com/Y2K-Finance/Earthquake/blob/c4e1b11fa209fa31c704c5cc309698a13b6a9842/src/Controller.sol#L185  
 
And when a triggerDepeg happens claim tvl is assigned to the counterparty vault.  

If you ran the tests you would see that this Test( https://github.com/Y2K-Finance/Earthquake/blob/cc2d44e5249fd217208f85946ba1f2c50500aec5/test/ControllerTest.t.sol#L69 ) holds true, and makes what I say correct

Reply:
`claimTVL` should not be classified as `premium` as premium would only be the counterparty in case of winning. 

Subtracting fees should be outsourced completely to the controller. There should not be any logic on the vaults regarding fees IMO.

Reply:
> We are only and always were collecting fees if premium was bigger than collat

@MiguelBits  this comment is factually threw me off. Indeed if claimTVL is not just the `premium` but the entire TVL of after the vault is resolved, calc the entitled shares is redundant and your prev approach is fine. 

However in general think it would be a better alternative to have the entire fee logic on the controller and set claimTVL (should be renamed to `epochResolvedTVL`) to `epochResolvedTVL - fee` therefore no need to calc fees on vault anymore. 

Also it's important to have old versions of the contracts as files in the repo, as they will still be active for a while, therefore there should be multiple folders `earthquake@1.0.0` and `earthquake@1.1.0`

Reply:
version forking and contract design choices are separate issues so marked as solved